module/mempool/stdmap/incorporated_result_seals.go

```diff
@@ -30,7 +30,7 @@ func NewIncorporatedResultSeals(limit uint) *IncorporatedResultSeals {
 		maxHeight := uint64(0)
 		var sealsAtMaxHeight sealSet
 		for height, seals := range byHeight {
-			if height > maxHeight {
+			if height > maxHeight || (height == 0 && maxHeight == 0) {
```

Comment:
I'm not sure how useful it is with this check. 

if the height of the seal is 0, then it's a seal for root block. However, the root seal will never go to mempool, as it is always included in the genesis block.

Reply:
Such data would indeed never be created by a honest node in its normal processing.

What guarantees have we built to be sure such data would not be processed otherwise? I'm asking because we're crashing the node upon seeing such data (a larger-than-limit number of incomplete seals for a block at height 0).

Reply:
>What guarantees have we built to be sure that kind of seal would not be processed

The block height will only be increasing. If we ever add a seal to this mempool, the seal must have been validated. In other words, an incomplete seal won't be added to seals mempool. Hence, a valid seal must have height greater than 0.